App.js

```diff
@@ -157,212 +154,271 @@ const routeConfigs = {
 
 export default class App extends Component {
 
-    input = null;
-
-    constructor(props) {
-      super(props);
-      this.state = {
-        client: null,
-        initialized: false,
-        loading: false,
-        settings: {},
-        server: null,
-        text: '',
-        user: null,
-        serverError: false,
-      }
+  input = null
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      client: null,
+      initialized: false,
+      loading: false,
+      settings: {},
+      server: null,
+      text: '',
+      user: null,
+      serverError: false,
+      unassign_notifs: [],
+      assign_notifs: [],
+      unassign_notifs_flush_timeout: null,
+      assign_notifs_flush_timeout: null
     }
 
-    componentWillMount() {
+    this.sendAssignNotifs = this.sendAssignNotifs.bind(this)
+    this.sendUnassignNotifs = this.sendUnassignNotifs.bind(this)
+  }
+
+  componentWillMount() {
+
+    Settings.addListener('server:remove', this.disconnect.bind(this))
+    Settings.addListener('websocket:message', this.onWebSocketMessage.bind(this))
+    Settings.addListener('user:login', (event) => {
+      const { client, user } = event
+      if (user && user.isAuthenticated() && (user.hasRole('ROLE_COURIER') || user.hasRole('ROLE_ADMIN'))) {
+        Registry.initWebSocketClient(client)
+      }
+    })
+    Settings.addListener('user:logout', () => Registry.clearWebSocketClient())
+
+    AppUser.load()
+      .then(user => {
+        Settings
+          .loadServer()
+          .then(baseURL => this.initializeRouter(baseURL, user))
+      })
+  }
+
+  onWebSocketMessage(event) {
```

Comment:
Good idea to have bufferized the notifications. 
But I think it is wrong to manage it through the state: each time you change the state, the component re-renders, but nothing has actually changed. 
I think the buffer should be extracted to a dedicated class.